test/models/users_test.rb

```diff
@@ -58,10 +59,88 @@ class UsersTest < ActiveSupport::TestCase
     end
 
     test "find_or_create_from_github accepts minimal users without name nor email" do
+      FakeWeb.register_uri(:get, "https://api.github.com/user/emails", status: %w(200 OK), body: [].to_json, content_type: "application/json")
       user = User.find_or_create_from_github(@minimal_github_user)
       assert_equal @minimal_github_user.login, user.login
     end
 
+    test "find_or_create_from_github selects any email when org email is unspecified" do
+      github_org_user = stub(
+        id: 42,
+        name: 'Jim Jones',
+        login: 'jim',
+        email: "jim@#{@org_domain}",
+        avatar_url: 'https://avatars.githubusercontent.com/u/42?v=3',
+        url: 'https://api.github.com/user/jim',
+      )
+
+      previous_preferred_org_emails = Shipit.preferred_org_emails
+
+      begin
+        Shipit.preferred_org_emails = [].freeze
+        user = User.find_or_create_from_github(github_org_user)
+        assert_equal github_org_user.email, user.email
+      ensure
+        Shipit.preferred_org_emails = previous_preferred_org_emails
+      end
+    end
+
+    test "find_or_create_from_github selects org email for user" do
+      Shipit.preferred_org_emails = [@org_domain]
+      expected_email = "myuser@#{@org_domain}"
+      FakeWeb.register_uri(:get, "https://api.github.com/user/emails", status: %w(200 OK), body: [{email: expected_email}].to_json, content_type: "application/json")
+
+      user = User.find_or_create_from_github(@github_user)
+      assert_equal expected_email, user.email
+    end
+
+    test "find_or_create_from_github selects private and primary org email for user when necessary" do
+      Shipit.preferred_org_emails = [@org_domain]
```

Comment:
That's global state, it need to be properly reset after you test. Best way to do it is:

```ruby
teardown do
  Shipit.preferred_org_emails = nil
end

Reply:
Right, I don't want to affect other tests. I used a `begin...ensure` block in another test to do this, but I'll strip it out in favor of setup/teardown to cover the whole test class.